src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -105,6 +106,29 @@ public Dataset find(Object pk) {
         return em.find(Dataset.class, pk);
     }
 
+    public Dataset findDeep(Object pk) {
+        return (Dataset) em.createNamedQuery("Dataset.findById")
+            .setParameter("id", pk)
+            .setHint("eclipselink.left-join-fetch", "o.files.ingestRequest")
+            .setHint("eclipselink.left-join-fetch", "o.files.thumbnailForDataset")
+            .setHint("eclipselink.left-join-fetch", "o.files.dataTables")
+            .setHint("eclipselink.left-join-fetch", "o.files.auxiliaryFiles")
+            .setHint("eclipselink.left-join-fetch", "o.files.ingestReports")
+            .setHint("eclipselink.left-join-fetch", "o.files.dataFileTags")
+            .setHint("eclipselink.left-join-fetch", "o.files.fileMetadatas")
+            .setHint("eclipselink.left-join-fetch", "o.files.fileMetadatas.fileCategories")
+            .setHint("eclipselink.left-join-fetch", "o.files.guestbookResponses")
+            .setHint("eclipselink.left-join-fetch", "o.files.embargo")
+            .setHint("eclipselink.left-join-fetch", "o.files.fileAccessRequests")
```

Comment:
@ErykKul  - FWIW - at QDR, I've had to remove this line in two different versions. QDR is a fork, but I think the key issue is probably that in once case, the code is ~5.13 which is before the FileAccessRequest class was created to add the request time (a recent PR) and in the other my db has an additional column on the fileaccessrequests table due to some experimenting on another PR (#9599). I don't think that affects this indexing PR directly but - does that make sense? other than performance, is it true that dropping this hint (or any hint?) shouldn't affect anything? Or is this unique because we're not using fileaccessrequests when indexing? Any thoughts welcome.

Reply:
The hints do result in improved performance because eclipse link picks them up and respects them. They do not influence functionality. Indexing does not use the FileAccessRequest, so its performance would not be affected. In the dataset page it will be, since it does use FileAccessRequest (this is why I have added the hint, I have noticed thousands of queries retrieving the FileAccessRequest objects before showing the page with a large dataset to the user).